app/gui/view/src/notification/api.rs

```diff
@@ -0,0 +1,580 @@
+pub struct UpdateOptions {
+    /// New state of the toast.
+    #[deref]
+    #[deref_mut]
+    #[serde(flatten)]
+    pub options: Options,
+    /// Used to update a toast content.
+    pub render:  Option<Content>,
+}
+
+impl_try_from_jsvalue! { UpdateOptions }
+
+impl UpdateOptions {
+    /// Allows set a new message in the notification.
+    ///
+    /// Useful only for update a toast.
+    pub fn raw_text_content(mut self, render: impl AsRef<str>) -> Self {
+        let message = render.as_ref();
+        self.render = Some(message.into());
+        self
+    }
+
+    /// Allows set a new message in the notification.
+    ///
+    /// This sets is as plain text (it will not be interpreted as HTML).
+    pub fn set_raw_text_content(&mut self, render: impl Into<String>) {
+        let content = Content::Text(render.into());
+        self.render = Some(content);
+    }
+}
+
+
+
+// ====================
+// === Notification ===
+// ====================
+
+/// A persistent notification.
+///
+/// It can be updated or dismissed.
+#[derive(Clone, CloneRef, Debug)]
+pub struct Notification {
+    /// The unique notification id. We use it to dismiss or update the notification.
+    id:         Rc<RefCell<Id>>,
+    /// The notification options. They will be used to show/update the notification.
+    options:    Rc<RefCell<UpdateOptions>>,
+    /// The options that we have used to show the notification the last time.
+    last_shown: Rc<RefCell<Option<UpdateOptions>>>,
+}
+
+impl Default for Notification {
+    fn default() -> Self {
+        Self::new(default())
+    }
+}
+
+impl From<Id> for Notification {
+    fn from(id: Id) -> Self {
+        let id = Rc::new(RefCell::new(id));
+        let options = default();
+        let last_shown = default();
+        Self { id, options, last_shown }
+    }
+}
+
+impl Display for Notification {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        Display::fmt(&*self.id.borrow(), f)
+    }
+}
+
+impl Notification {
+    /// Create a new notification archetype.
+    ///
+    /// It will not be shown until you call [`Notification::show`].
+    pub fn new(options: UpdateOptions) -> Self {
+        let id = Id::new_unique();
+        let id = Rc::new(RefCell::new(id));
+        let options = Rc::new(RefCell::new(options));
+        let last_shown = default();
+        Self { id, options, last_shown }
+    }
+
+    fn update_inner(&self, options: UpdateOptions) -> Result<(), JsValue> {
+        self.options.replace(options);
+        let ret = self.id.borrow().update(&self.options.borrow());
+        if ret.is_ok() {
+            let new_options = self.options.borrow().clone();
+            // If the ID was updated, we need to keep track of it.
+            if let Some(new_id) = new_options.toast_id.as_ref() {
+                self.id.replace(new_id.as_str().into());
+            }
+            self.last_shown.replace(Some(new_options));
+        }
+        ret
+    }
```

Comment:
Is there a reason why it's not a public method called "set_options"?

Reply:
No, other than I haven't needed.
Refactored.